README.md

```diff
@@ -396,14 +396,71 @@ authClient.signIn({
 
 ### `signOut()`
 
-Signs the user out of their current Okta [session](https://developer.okta.com/docs/api/resources/sessions#example).
+Signs the user out of their current Okta [session](https://developer.okta.com/docs/api/resources/sessions#example) and clears all tokens stored in the `TokenManager`.
+
+By default:
+
+  - the user will be signed out using an API method 
+  - the page location will not be changed. 
+  
+When calling `signOut` with the default options, it is recommended to reload the page after the promise resolves to ensure all user data is cleared from memory and the app is properly loaded in an unauthenticated state.
+
+Since the API method relies upon an XHR call and the ability to set 3rd-party cookies, it may fail in certain scenarios, leaving the user's remote session open. For this reason, we generally recommend using the redirect method over the API method. The redirect method sets the window location to Okta where the session will be fully cleared.
+
+`signOut` takes the following options:
+
+* `redirect` - If `true`, enables the logout redirect method. By default, this will redirect the user to an Okta-hosted login page. Logging back in from this page may send the user to their Okta user homepage rather than the application.
+* `redirectUri` - Enables logout redirect using a custom URI to which the browser will be redirected after logout. This URI must be whitelisted as a `Logout Redirect Uri` in your application's general settings. It is common to use your application URI here.
+* `idToken` - Enables logout redirect and specifies the idToken. By default, `signOut` will look for a token named `idToken` within the `TokenManager`. If you have stored the idToken in a different location, you should retrieve it first and then pass it here. Passing a value of `false` for `idToken` will disable all logic with the `idToken`. In any case where the `idToken` is not available, the user will be signed out using the API method.
```

Comment:
This is a bit dense.  (or maybe I just am)

If I say `redirect: false` and `idToken: "asdfasdf"`, will I use redirect or not?   If say `redirect: true` and dont' specify `idToken`, will things "just work" if I haven't stored the token elsewhere?  If I say `redirect: true` and don't specify `idToken`, but it's not in the default place, it will silently act as if `redirect: false`?  

This seems easy to mess up.

Reply:
Generally, nobody should need to set the idToken unless they are doing something exotic. We read it from the TokenManager for either `redirect` or `redirectUri`.  

The only difference between these two options is whether you want to redirect to Okta or to a whitelisted URI.

Passing `idToken` without one of the other redirect options will implicitly enable redirect (to Okta). Maybe this is the source of confusion? We could make it so that `idToken` is ignored unless one of the 2 redirect options is chosen.

Reply:
Yeah, I think requiring it to be explicitly set would reduce the potential confusion.  

What about the case where `redirect: true` and the `idToken` can't be found (I realize that should be unusual, but...) . Wouldn't it be better to give an explicit error rather than silently switching to `redirect: false`?

Reply:
The case of missing an idToken is not unusual. The idToken will be cleared by the TokenManager if an error is encountered. So it may happen for some percentage of user sessions that the idToken is missing for natural reasons. So in all cases we are handling the loss of the idToken by falling back to API method and then doing the redirect ourselves.

Reply:
I removed the case of `idToken` *implicitly* turning on the redirect option. You will have to pass either `redirect` or `postLogoutRedirectUri`